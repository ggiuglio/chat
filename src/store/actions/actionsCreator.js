import {
  TEST_ACTION
} from './actionsTypes.js'
 
export const testAction = () => {
  return dispatch => {
    dispatch({
      type: TEST_ACTION
    });
  }
}