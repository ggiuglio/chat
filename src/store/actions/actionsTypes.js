export const LOAD_MESSAGES = "LOAD_MESSAGES";
export const WRITE_MESSAGE = "WRITE_MESSAGE";