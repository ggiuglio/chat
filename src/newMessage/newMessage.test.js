import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { loadMessagesMock } from '../store/actions/actionsCreator';
import ReactTestUtils from 'react-dom/test-utils';


import NewMessage from './newMessage';
const { act } = renderer;


const mockStore = configureStore([]);

describe('New Message', () => {
  let store;
  let component;
  const stateMessages = [
    {
      "author": "Tom",
      "id": 0,
      "isMine": false,
      "message": "Hi this is my first message",
      "timeStamp": "1/19/1970",
    },
    {
      "author": "Lisa",
      "id": 1,
      "isMine": false,
      "message": "How is the project going?",
      "timeStamp": "1/19/1970",
    },
    {
      "author": "Tom",
      "id": 2,
      "isMine": false,
      "message": "It proceeds fine?",
      "timeStamp": "1/19/1970",
    },
    {
      "author": "Giulio",
      "id": 3,
      "isMine": true,
      "message": "Hi, can I have access to the shared folder?",
      "timeStamp": "1/19/1970",
    },
  ];

  beforeEach(() => {
    store = mockStore({
      messages: stateMessages
    });
    store.dispatch = jest.fn();
    act(() => {
      component = renderer.create(
        <Provider store={store}>
          <NewMessage />
        </Provider>
      );
    });
  });

  it('should render a message', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

});