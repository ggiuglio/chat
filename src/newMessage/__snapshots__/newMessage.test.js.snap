// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Message should render a message 1`] = `
<div
  className="sc-bdVaJa ejpxlp"
>
  <textarea
    className="sc-bwzfXH bShnlp"
    onChange={[Function]}
    value=""
  />
  <button
    className="sc-htpNat iltKDN"
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
